packages/eslint/src/rules/html/migrations/no-deprecated-loader.ts

```diff
@@ -1,56 +1,29 @@
-import { createRule } from '../../../utils/create-rule';
-import { HtmlNode } from '../../../parsers/html/html-node';
+import { createClassUpdateRule } from '../../../utils/create-class-update-rule';
+import { generateReplacedClassMutations } from '../../../utils/generate-mutations';
+import { generateReplacedClassMessages } from '../../../utils/generate-messages';
 
 export const name = 'no-deprecated-loader';
 
-// Type: RuleModule<"uppercase", ...>
-export default createRule({
-  name,
-  meta: {
-    docs: {
-      dir: 'html',
-      description:
-        'Flags deprecated "loader-xs" and "loader-sm" classes and replaces them with "loader-16" and "loader-40" respectively.',
-    },
-    messages: {
-      deprecatedLoaderXS:
-        'The "loader-xs" class is deprecated. Please replace it with "loader-16".',
-      deprecatedLoaderSM:
-        'The "loader-sm" class is deprecated. Please replace it with "loader-40".',
-    },
-    type: 'problem',
-    fixable: 'code',
-    schema: [],
-  },
-  defaultOptions: [],
-  create(context) {
-    return {
-      tag(node: HtmlNode) {
-        const $node = node.toCheerio();
-
-        if ($node.hasClass('loader-xs')) {
-          context.report({
-            messageId: 'deprecatedLoaderXS',
-            loc: node.loc,
-            fix(fixer) {
-              const fixedNode = $node.removeClass('loader-xs').addClass('loader-16');
-              return fixer.replaceTextRange(node.range, fixedNode.toString());
-            },
-          });
-        }
+const classesMap = [
+  { old: 'loader', new: 'spinner' },
+  { old: 'loading-modal', new: 'spinner-modal' },
+  { old: 'loader-xs', new: 'spinner-16' },
+  { old: 'loader-sm', new: 'spinner-40' },
```

Comment:
There is already a rule that turns `loader-[xs|sm]` into `spinner-16|40` : https://github.com/swisspost/design-system/blob/main/packages/eslint/src/rules/html/migrations/no-deprecated-loader.ts
It would act as a "Phase 1".